aiida/engine/processes/calcjobs/tasks.py

```diff
@@ -270,15 +273,65 @@ async def do_retrieve():
         )
     except (plumpy.futures.CancelledError, plumpy.process_states.Interruption):  # pylint: disable=try-except-raise
         raise
-    except Exception:
+    except Exception as exception:
         logger.warning(f'retrieving CalcJob<{node.pk}> failed')
-        raise TransportTaskException(f'retrieve_calculation failed {max_attempts} times consecutively')
+        raise TransportTaskException(f'retrieve_calculation failed {max_attempts} times consecutively') from exception
     else:
-        node.set_state(CalcJobState.PARSING)
+        node.set_state(CalcJobState.DONE)
         logger.info(f'retrieving CalcJob<{node.pk}> successful')
         return result
 
 
+async def task_stash_job(node: CalcJobNode, transport_queue: TransportQueue, cancellable: InterruptableFuture):
+    """Transport task that will optionally stash files of a completed job calculation on the remote.
+
+    The task will first request a transport from the queue. Once the transport is yielded, the relevant execmanager
+    function is called, wrapped in the exponential_backoff_retry coroutine, which, in case of a caught exception, will
+    retry after an interval that increases exponentially with the number of retries, for a maximum number of retries.
+    If all retries fail, the task will raise a TransportTaskException
+
+    :param node: the node that represents the job calculation
+    :param transport_queue: the TransportQueue from which to request a Transport
+    :param cancellable: the cancelled flag that will be queried to determine whether the task was cancelled
+    :type cancellable: :class:`aiida.engine.utils.InterruptableFuture`
+    :raises: Return if the tasks was successfully completed
+    :raises: TransportTaskException if after the maximum number of retries the transport task still excepted
+    """
+    if node.get_state() == CalcJobState.RETRIEVING:
+        logger.warning(f'calculation<{node.pk}> already marked as RETRIEVING, skipping task_stash_job')
+        return
+
+    initial_interval = get_config_option(RETRY_INTERVAL_OPTION)
+    max_attempts = get_config_option(MAX_ATTEMPTS_OPTION)
+
+    authinfo = node.computer.get_authinfo(node.user)
+
+    async def do_stash():
+        with transport_queue.request_transport(authinfo) as request:
+            transport = await cancellable.with_interrupt(request)
+
+            logger.info(f'stashing calculation<{node.pk}>')
+            return execmanager.stash_calculation(node, transport)
+
+    try:
+        await exponential_backoff_retry(
+            do_stash,
+            initial_interval,
+            max_attempts,
+            logger=node.logger,
+            ignore_exceptions=plumpy.process_states.Interruption
+        )
+    except plumpy.process_states.Interruption:
+        raise
```

Comment:
What kind of exception is this catching? Can't we re-raise something more informative?

Reply:
is catching what it says its catching 😜 an interruption like pausing or killing, so no

Reply:
Right, but my underlying question is why would you catch the exception to just raise with nothing? Why not just leave `plumpy.process_states.Interruption` be the raising error?

Reply:
Just in case this what your question was trying to address: if you catch and then simply call `raise` again, it will raise the original exception. Why is this useful? Because this way you can treat certain exceptions different from others. In this case we want exceptions to be treated a certain way, except the `Interruption`, which should just be bubbled up.

Reply:
> if you catch and then simply call `raise` again, it will raise the original exception.

Ahh ok. Yeah, sorry, I did not know this, I thought you had to explicitly provide the exception to raise.